app/controllers/my-projects/project/recommendations/add.js

```diff
@@ -25,58 +39,80 @@ export default class MyProjectsProjectRecommendationsAddController extends Contr
-      ...this.recommendationOptions.disapproved.actions,
-      ...this.recommendationOptions.get('disapproved-with-modifications-conditions').actions,
-      ...this.recommendationOptions.get('not-available').actions,
-    ];
-    return allActions;
-  }
+  // recommendation options for disposition.communityboardrecommendation,
+  // disposition.boroughboardrecommendation, disposition.boroughpresidentrecommendation
+  recOptions = [
+    'Approved',
+    'Approved with Modifications/Conditions',
+    'Disapproved',
+    'Disapproved with Modifications/Conditions',
+    'Waived',
+  ];
 
   @action
   setProp(property, newVal) {
     this.set(property, newVal);
   }
 
+  // For setting disposition recommendation, use the action
+  // "setDispositionRecByPartType" instead.
   @action
   updateDispositionAttr(disposition, attrName, newVal) {
     disposition.set(attrName, newVal);
   }
+
+  /**
+ * @param { Disposition } disposition
+ * @param { String } recommendation
+ * @param { String } participantType (optional)
+ * assigns the `recommendation` string to either the `
+ * TODO: Update this to rely on the disposition.participantType field
+ * when it is implemented in the ZAP-API.
+ */
+  @action
+  setDispositionRecByPartType (disposition, recommendation) {
+    const partType = this.participantType;
+    let recommendationFieldName = '';
+    if (partType === 'CB') {
+      recommendationFieldName = 'communityboardrecommendation';
+    } else if (partType === 'BB') {
+      recommendationFieldName = 'boroughboardrecommendation';
+    } else if (partType === 'BP') {
+      recommendationFieldName = 'boroughpresidentrecommendation';
+    } else {
+      console.log('ZAP Error: Invalid disposition participant type.');
+      this.transitionToRoute('oops');
+    }
+    disposition.set(recommendationFieldName, recommendation);
+  }
+
+  /**
+   * Saves all dispositions on the loaded project with user-inputted recommendation values.
+   * If 'allActions" is `True`, will copy values in dispositionForAllActions into each
+   * disposition before saving those dispositions.
+   */
+  @action
+  submitRecommendations() {
+    const thisCtrl = this;
+    this.dispositions.forEach(function(disposition) {
+      if (thisCtrl.allActions) {
+        thisCtrl.send('setDispositionRecByPartType', disposition, thisCtrl.dispositionForAllActions.recommendation);
+        disposition.set('votinginfavorrecommendation', thisCtrl.dispositionForAllActions.votinginfavorrecommendation);
+        disposition.set('votingagainstrecommendation', thisCtrl.dispositionForAllActions.votingagainstrecommendation);
+        disposition.set('votingabstainingonrecommendation', thisCtrl.dispositionForAllActions.votingabstainingonrecommendation);
+        disposition.set('totalmembersappointedtotheboard', thisCtrl.dispositionForAllActions.totalmembersappointedtotheboard);
+        disposition.set('consideration', thisCtrl.dispositionForAllActions.consideration);
+      }
```

Comment:
You can do this with setProperties and it'd be much more legible. If you're concerned about using POJOs in the future, you can import setProperties from `@ember/object` and use it like this:
```
setProperties(disposition, {
  // ...
});
```

Reply:
✅